epub33/core/index.html

```diff
@@ -6082,9 +6082,90 @@ <h4>URLs in the OCF Abstract Container</h4>
+						</li>
+
+						<li>
+							Let <var>testURLStringA</var> be the result of applying the <a data-cite="url#concept-url-serializer">URL Serializer</a> to <var>testURLRecord</var>.
+						</li>
+
+						<li>
+							Set the <a>container root URL</a> to <code>https://b.example.org/B/</code>.
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">The reasons to repeat the same steps twice with different, and artifical, settings of the container root URL is to avoid collision which may occur if the <var>url</var> string also includes <code>/A/</code>. Consider, for example, the case where <var>url</var> is of the form <code>"../../A/doc.xhtml</code>.</p>
+							</details>
+						</li>
+
+						<li>
+							Let <var>base</var> be the <a data-cite="url#concept-base-url">base URL</a> that must be used to <a data-cite="url#concept-url-parser">parse</a> <var>url</var> as defined by the context (document or environment) where <var>url</var> is used, and according to the <a>container root URL</a>. 
+						</li>
+
+						<li>
+							Let <var>testURLRecord</var> be the result of applying the <a data-cite="url#concept-url-parser">URL parser</a> to <var>url</var>, with <var>base</var>.
+						</li>
+
+						<li>
+							Let <var>testURLStringB</var> be the result of applying the <a data-cite="url#concept-url-serializer">URL Serializer</a> to <var>testURLRecord</var>.
+						</li>
+
+						<li>
+							If <var>testURLStringA</var> does not start with <code>https://a.example.org/</code> or <var>testURLStringB</var> does not start with <code>https://b.example.org/</code>, return <var>true</var>.
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">If any of the result does not share the test URL host, it means the test URL, or the <code>base</code> element, was <em>absolute</em> and point outside the container. This is acceptable.</p>
+							</details>
+						</li>
+
+						<li>
+							If <var>testURLStringA</var> starts with <code>https://a.example.org/A/</code> and <var>testURLStringB</var> starts with <code>https://b.example.org/B/</code>, return <var>true</var>.
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">The presence of the first test path segments (<code>A</code>, respectively <code>B</code>) indicate that the URL doesn't leak outside the container.</p>
+							</details>
+						</li>
+
+						<li>Return <var>false</var>.</li>
+					</ol>
+
+					<div class="note">
+						<p>	
+							The properties of the <a>container root URL</a> are such that whatever the amount
+							of <a data-cite="url/#double-dot-path-segment">double-dot path segments</a> in a URL string (for
+							example, <code>../../../secret</code>), it is parsed to a <a>content URL</a> (and not "leak"
+							outside the container). This avoids potential run-time security issues. However, the additional
```

Comment:
What if we moved this note before the algorithm, to introduce why we define this extra requirement (the algorithm)?

Reply:
Not sure. I am a bit bothered if there is a note preceding what is being formally defined; it somehow breaks the (normative) narrative for me. I guess it is a bit of a taste issue; why don't I leave it as is, and see how others react?

Reply:
Sure, works for me!

Reply:
Thinking about it further, and also related to your note on adding an explanation to the start of the algorithm, I agree and have moved the first paragraph of the note.